src/EllipseCreation.ts

```diff
@@ -55,44 +57,64 @@ function drawEllipse(original: Point, current: Point) {
 
     ctx.beginPath();
     ctx.ellipse(center.x, center.y, rx, ry, Math.PI / 2, 0, 2 * Math.PI);
+    //I know this is stupid to constantly make a new ellipse but my brain hurts I'm sorry
     ctx.stroke();
+    currentEllipse = new Ellipse(center, rx, ry);
+    return currentEllipse;
 }
 
 /**
  * A function to begin a mode to draw cuts.
  * If atomMode was previously active, remove the listener.
  */
-function ellipseMode() {
+export function ellipseCreation() {
     canvas.addEventListener("mousedown", mouseDown);
 }
-(window as any).ellipseMode = ellipseMode;
 
 /**
  * Logs the position where the mouse is first pressed down. Begins the event for moving
  * the mouse, then ends it once mouseup occurs.
  * @param event The even of holding down the mouse
  */
 function mouseDown(event: MouseEvent) {
-    startingPoint = {x: event.clientX - xshift, y: event.clientY - yshift};
+    startingPoint = {x: event.clientX, y: event.clientY};
     canvas.addEventListener("mousemove", mouseMoving);
-    canvas.addEventListener("mouseup", () => {
-        canvas.removeEventListener("mousemove", mouseMoving);
-    });
+    canvas.addEventListener("mouseup", mouseUp);
 }
 
 /**
  * Follows the current mouse position, and draws the ellipse between the starting
  * point and the current point.Clears the canvas with every movement. When the
  * tree is finished a redraw function will be made to draw all of the already
  * created cuts and atoms.
- * @param event The even of a mouse moving
+ * @param event The event of a mouse moving
  */
 function mouseMoving(event: MouseEvent) {
     //As strange as this is, this is the only way to clear the canvas
     ctx.clearRect(0, 0, canvas.width, canvas.height);
     const currentPoint: Point = {
-        x: event.clientX - xshift,
-        y: event.clientY - yshift,
+        x: event.clientX,
+        y: event.clientY,
     };
-    drawEllipse(startingPoint, currentPoint);
+    redrawCut(tree.sheet);
+    currentEllipse = createEllipse(startingPoint, currentPoint);
+}
+
+/**
+ * When the mouse is lifted up, removes the movement listener and adds it to the tree itself.
+ */
+function mouseUp() {
+    canvas.removeEventListener("mousemove", mouseMoving);
+    const newCut: CutNode = new CutNode(currentEllipse);
+    if (tree.canInsertAEG(newCut, currentEllipse.center)) {
+        tree.insertAEG(newCut, currentEllipse.center);
+    }
+    canvas.removeEventListener("mouseup", mouseUp);
```

Comment:
Id prefer you keep the together so its more clear. 
```suggestion
    canvas.removeEventListener("mousemove", mouseMoving);
    canvas.removeEventListener("mouseup", mouseUp);
    const newCut: CutNode = new CutNode(currentEllipse);
    if (tree.canInsertAEG(newCut, currentEllipse.center)) {
        tree.insertAEG(newCut, currentEllipse.center);
    }
```